subsys/net/lib/sockets/sockets.c

```diff
@@ -1134,6 +1134,54 @@ static int sock_get_pkt_src_addr(struct net_pkt *pkt,
 	return ret;
 }
 
+#if defined(CONFIG_NET_OFFLOAD)
+static int sock_get_offload_pkt_src_addr(struct net_pkt *pkt,
+					 struct sockaddr *addr,
+					 socklen_t addrlen)
+{
+	int ret = 0;
+
+	if (!addr || !pkt) {
+		return -EINVAL;
+	}
+
+	if (IS_ENABLED(CONFIG_NET_IPV4) &&
+	    net_pkt_family(pkt) == AF_INET) {
+		if (addrlen < sizeof(struct sockaddr_in)) {
+			ret = -EINVAL;
+			goto error;
+		}
+
+		memcpy(addr, &pkt->remote,
+		       MIN(addrlen, sizeof(pkt->remote)));
+	} else {
```

Comment:
It is trivial to support IPv6 too so we need to have it here. So something like
```
	} else if (IS_ENABLED(CONFIG_NET_IPV6) && net_pkt_family(pkt) == AF_INET6) {
		if (addrlen != sizeof(struct sockaddr_in6)) {
			ret = -EINVAL;
			goto error;
		}

		memcpy(addr, &pkt->remote, MIN(addrlen, sizeof(pkt->remote)));
	} else {
```

Reply:
Added